addons/project/models/project_project.py

```diff
@@ -413,6 +413,11 @@ def map_tasks(self, new_project_id):
         subtasks_not_displayed = all_subtasks.filtered(
             lambda task: not task.display_in_project
         )
+        new_tasks._get_all_subtasks().filtered(
+            lambda child: child.project_id == self
+        ).write({
+            'project_id': project.id
+        })
         project.write({'tasks': [Command.set(new_tasks.ids)]})
```

Comment:
I would expect that this is equivalent to:

```suggestion
        new_tasks_and_subtasks =  new_tasks + all_subtasks.filtered_domain([
            ('project_id', '!=', self.id)
        ])
        project.tasks = new_tasks_and_subtasks
```
(since `project.project().tasks` it the One2many field corresponding to `project.task().project_id`)